embassy-stm32/src/usart/mod.rs

```diff
@@ -241,6 +240,29 @@ impl<'d, T: BasicInstance, TxDma, RxDma> Uart<'d, T, TxDma, RxDma> {
         }
     }
 
+    pub fn new_with_rtscts(
+        _inner: impl Peripheral<P = T> + 'd,
+        rx: impl Peripheral<P = impl RxPin<T>> + 'd,
+        tx: impl Peripheral<P = impl TxPin<T>> + 'd,
+        rts: impl Peripheral<P = impl RtsPin<T>> + 'd,
+        cts: impl Peripheral<P = impl CtsPin<T>> + 'd,
+        tx_dma: impl Peripheral<P = TxDma> + 'd,
+        rx_dma: impl Peripheral<P = RxDma> + 'd,
+        config: Config,
+    ) -> Self {
+        into_ref!(cts, rts);
+
+        unsafe {
+            rts.set_as_af(rts.af_num(), AFType::OutputPushPull);
+            cts.set_as_af(cts.af_num(), AFType::Input);
+            T::regs().cr3().write(|w| {
```

Comment:
Are you sure this works? `new()` is doing RCC reset+enable *after* this write, so it looks a bit suspicious.

Reply:
Hmm, I am not sure that I see how RCC reset and enable alters the UART control regs?

Reply:
I will be able to verify this on monday though.

Reply:
Writes to peripheral regs simply don't work if the peripehral is not rcc-enabled, and the rcc-reset clears all the registers.